source/snes/metroid3/rom.py

```diff
@@ -655,6 +655,10 @@ def _apply_improvements(self):
 														[0x07,0x01,0xED,0xFD,0xED,0xFC,0xED,0xFC,0xED,0xFD,0xED,0xFC,0xED,0xFC], "1"*14)
 		self._apply_single_fix_to_snes_address(0x90CBB3,[0x07,0x01,0x07,0xFD,0x07,0xFC,0x07,0xFC,0x07,0xFD,0x07,0xFC,0x07,0xFC],
 														[0x02,0x01,0x0B,0xFD,0x0B,0xFC,0x0B,0xFC,0x0B,0xFD,0x0B,0xFC,0x0B,0xFC], "1"*14)
+		
+		#makes samus be able to breathe after she ran into a wall
+		self._apply_single_fix_to_snes_address(0x91B010+0x89*2,0xB2B4,0xB298,2)
+		self._apply_single_fix_to_snes_address(0x91B010+0x8A*2,0xB2B4,0xB298,2)
```

Comment:
Makes sense to add these lines here when introducing a new routine/hook/bypass

Reply:
The relocation is shifting them from their vanilla durations: 0x10, 0xFF which is a 16-frame animation followed by a command to loop back to the beginning of the 16-frame animation.
The new duration array would be the same as the normal standing animations, which have four 10-frame animations followed by a low-health conditional loop which dovetails into a set of 8-frame animations and a loopback.

Reply:
I don't see any immediate impact on the game behavior with the change in this file and the one above it; they seem to be an attempt to prevent the player animation from glitching/doing a bad VRAM load in the case that a player should run into a wall while other "glitchy things" like grapple or MB stand-up glitch are happening that might advance the animation index past 0.  I would be curious to know which bug triggered the need for this change, because many players will want to make sure that their usual glitches still work the "same way", so are we fixing a vanilla bug or are we making SpriteSomething work more like the original vanilla game?